pkg/webhook/admission/pod/agent_injector.go

```diff
@@ -196,23 +197,59 @@ func (ag *AgentInjector) InjectAgent(pod *v1.Pod) error {
 	}
 
 	queueProxyEnvs := []v1.EnvVar{}
-	queueProxyAvailable := false
+	containerPort := constants.InferenceServiceDefaultHttpPort
+	readinessProbeAvailable := false
+
 	for _, container := range pod.Spec.Containers {
 		if container.Name == "queue-proxy" {
 			queueProxyEnvs = container.Env
-			queueProxyAvailable = true
+		}
+
+		if container.Name == "kserve-container" {
+			containerPort = fmt.Sprint((container.Ports[0].ContainerPort))
 		}
 	}
 
-	if !queueProxyAvailable {
-		readinessProbeJson, err := json.Marshal(pod.Spec.Containers[0].ReadinessProbe)
-		if err != nil {
-			return err
+	containerReadinessProbe := &v1.Probe{
```

Comment:
Why do we need to define the readiness probe here? 
- For serverless mode readiness probe can be specified on kserve container and queue proxy gets that via Env `SERVING_READINESS_PROBE`.
- For native mode,  we inject the `SERVING_READINESS_PROBE` env for model agent container.

Reply:
@andyi2it Can you help remind me why we still need to define the readiness probe here ?

Reply:
so the env is injected correctly with the port already 
```yaml
    - name: USER_PORT
      value: "5000"
    - name: SERVING_READINESS_PROBE
      value: '{"httpGet":{"path":"/v1/models/hello-lambda","port":5000,"host":"127.0.0.1","scheme":"HTTP","httpHeaders":[{"name":"K-Kubelet-Probe","value":"queue"}]},"successThreshold":1}'
```